File: tools/dbench/answers/go_sonnet/s2/e1_parse_fallback.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	values := []string{"12", "x", "7", "-", "30"}
	sum := 0
	bad := 0
	for _, s := range values {
		n, err := strconv.Atoi(s)
		if err != nil {
			bad++
			continue
		}
		sum += n
	}
	fmt.Println(sum)
	fmt.Println(bad)
}

File: tools/dbench/answers/rust_haiku/s3/e1_parse_fallback.rs
fn main() {
    let nums = ["12", "x", "7", "-", "30"];
    let mut sum = 0;
    let mut bad = 0;

    for s in &nums {
        match s.parse::<i32>() {
            Ok(n) => sum += n,
            Err(_) => bad += 1,
        }
    }

    println!("{}", sum);
    println!("{}", bad);
}
